src/EduManagementLab.Core/Services/CourseService.cs

```diff
@@ -68,5 +68,34 @@ public void DeleteCourse(Guid id)
             _unitOfWork.Courses.Remove(course);
             _unitOfWork.Complete();
         }
+        public Course CreateCourseMembership(Guid courseId, Guid userId, DateTime enrolledDate)
+        {
+            var course = GetCourse(courseId);
+
+            if (!course.Memperships.Any(c => c.UserId == userId))
+            {
+                course.Memperships.Add(new Course.Membership()
+                {
+                    Id = Guid.NewGuid(),
```

Comment:
Dont set Id. Entity Framework will give it an Id.

Reply:
Ohh right, i forgot that Entity is auto generate Id.